addons/website_profile/controllers/main.py

```diff
@@ -37,13 +37,17 @@ def _check_avatar_access(self, user_id, **post):
 
     def _check_user_profile_access(self, user_id):
         user_sudo = request.env['res.users'].sudo().browse(user_id)
+        denial_reason = False
         # User can access - no matter what - his own profile
         if user_sudo.id == request.env.user.id:
-            return user_sudo
-        if user_sudo.karma == 0 or not user_sudo.website_published or \
-            (user_sudo.id != request.session.uid and request.env.user.karma < request.website.karma_profile_min):
-            return False
-        return user_sudo
+            return user_sudo, denial_reason
+        if request.env.user.karma < request.website.karma_profile_min:
+            denial_reason = _("Not have enough karma to view other users' profile.")
```

Comment:
we can early return
(same below)
```suggestion
            return False, _("Not have enough karma to view other users' profile.")